df_py/volume/reward_calculator.py

```diff
@@ -0,0 +1,443 @@
+          chain where rewards go.
+        """
+        rewardsperlp: dict = {}
+        rewardsinfo: dict = {}
+
+        for i, LP_addr in enumerate(self.LP_addrs):
+            for j, (chainID, nft_addr) in enumerate(self.chain_nft_tups):
+                assert self.R[i, j] >= 0.0, self.R[i, j]
+                if self.R[i, j] == 0.0:
+                    continue
+
+                if chainID not in rewardsperlp:
+                    rewardsperlp[chainID] = {}
+                if LP_addr not in rewardsperlp[chainID]:
+                    rewardsperlp[chainID][LP_addr] = 0.0
+                rewardsperlp[chainID][LP_addr] += self.R[i, j]
+
+                if chainID not in rewardsinfo:
+                    rewardsinfo[chainID] = {}
+                if nft_addr not in rewardsinfo[chainID]:
+                    rewardsinfo[chainID][nft_addr] = {}
+                rewardsinfo[chainID][nft_addr][LP_addr] = self.R[i, j]
+
+        return rewardsperlp, rewardsinfo
+
+    @freeze_attributes
+    @enforce_types
+    def _get_chain_nft_tups(self) -> List[Tuple[int, str]]:
+        """
+        @return
+          chain_nft_tups -- list of (chainID, nft_addr), indexed by j
+        """
+        chainIDs = list(self.stakes.keys())
+        nft_addrs = self._get_nft_addrs()
+        chain_nft_tups = [
+            (chainID, nft_addr)  # all (chain, nft) tups with stake
+            for chainID in chainIDs
+            for nft_addr in nft_addrs
+            if nft_addr in self.stakes[chainID]
+        ]
+        return chain_nft_tups
+
+    @freeze_attributes
+    @enforce_types
+    def _get_nft_addrs(self) -> List[str]:
+        """
+        @return
+          nft_addrs -- list of unique nft addrs. Order is consistent.
+        """
+        nft_addrs = set()
+        for chainID in self.nftvols_USD:
+            for nft_addr in self.nftvols_USD[chainID]:
+                nft_addrs.add(nft_addr)
+
+        return sorted(nft_addrs)
+
+    @freeze_attributes
+    @enforce_types
+    def _get_lp_addrs(self) -> List[str]:
+        """
+        @return
+          LP_addrs -- list of unique LP addrs. Order is consistent.
+        """
+        LP_addrs = set()
+        for chainID in self.stakes:
+            for nft_addr in self.stakes[chainID]:
+                for LP_addr in self.stakes[chainID][nft_addr]:
+                    LP_addrs.add(LP_addr)
+
+        return sorted(LP_addrs)
+
+    @freeze_attributes
+    @enforce_types
+    def _get_predictoor_contracts(self) -> Dict[int, List[str]]:
+        """
+        @return
+          predictoors -- dict of (chainID, list of nft_addrs)
+
+        @notes
+          The keys in DEPLOYER_ADDRS are the chain ids
+          where predictoor contracts are deployed.
+          There is no point in query-ing on other chains
+        """
+        chainIDs = list(self.stakes.keys())
+        predictoors: Dict[int, List[str]] = {chain_id: [] for chain_id in chainIDs}
```

Comment:
Recall "predictoors" and "prediction contracts" are very different things. Predictoors are the people making predictions (or their bots). Predictoor contracts are the ERC20 datatoken contracts for prediction feeds.

You've already renamed from `get_predictoors()` to `get_predictoor_contracts()`. Great!

I guess I could have been more explicit about the other changes needed. They include:
- In `get_predictoor_contracts()` code: variable `predictoors` -> `predictoor_contracts`
- In `get_predictoor_contracts()` docstring: variable `predictoors` -> `predictoor_contracts`
- Attribute `self.predictoors` -> `self.predictoor_contracts`
- More? Please check